library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -444,7 +444,7 @@ private NetworkState checkSizeAllowedForNetwork(int networkType) {
      */
     public boolean isReadyToDownload() {
         synchronized (this) {
-            return isDownloadManagerReadyToDownload() && isClientReadyToDownload();
+            return isClientReadyToDownload() && isDownloadManagerReadyToDownload();
```

Comment:
@blundell can you elaborate?

The idea is that the client might have business logic that has nothing to do with the downloading by itself and so should fail/checked first. For example rooted device or user not logged in.